functions/assert_metadata.pp

```diff
@@ -0,0 +1,95 @@
+# Fails a compile if the client system is not compatible with the module's
+# `metadata.json`
+#
+# @param module_name
+#   The name of the module that should be checked
+#
+# @param options
+#   Behavior modifiers for the function
+#   * Can be set using `simplib::assert_metadata::options` in the `lookup`
+#     stack
```

Comment:
If I use this in a bunch of modules in my environment, and I want to test just one of them against an unsupported configuration, would setting this option disable it for everything?